minpy/dispatch/rule.py

```diff
@@ -253,3 +268,37 @@ def _get_arg_rule_key(self, args, kwargs):
         arg_key = [self._get_type_signiture(x) for x in args]
         kwarg_key = sorted(kwargs.keys())
         return '-'.join(arg_key) + '+' + '-'.join(kwarg_key)
+
+    @classmethod
```

Comment:
This is a little bit weird here. You take a `cls` argument and use the `_rules` belongs to it. This means one might create multiple policy objects and they have different rules associate with it, which is quite counter-intuitive. I think it is better to have only one policy object for one policy type. When we switch the policy, we are not constructing an object, but instead:
```python
policy.set_global_policy("only_numpy")
# policy.set_global_policy("auto_blacklist")
# policy.set_global_policy("prefer_mxnet")
```
This will also avoid importing the policy python file. What do you think?